models/cv_model.py

```diff
@@ -0,0 +1,214 @@
+# Note: Here we freeze the feature extractor, and only tune the MLC.
+# We used the default learning rate of Adam to train the MLC.
+
+# https://arxiv.org/pdf/2004.12274.pdf finetunes the feature extractor
+# on the ChestX-ray 14 dataset.
+from typing import Dict, Any
+import os.path as osp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.models as tvm
+
+# Texar Library
+from texar.torch import ModuleBase
+
+
+class SimpleFusionEncoder(ModuleBase):
+    r"""Visual feature extractor. Implementation is adapted
+    from
+    https://gitlab.int.petuum.com/shuxin.yao/image_report_generation/blob/master/implementation/Encoders/Encoders.py
+    Base encoder is set to be DenseNet121. Pretrained weights
+    are reused from https://github.com/berneylin/chexnet
+
+    NOTE: The output features are not a vector. Instead, we
+    treat the output from the feature layer of the densenet
+    as the features, and reshape it [batch size, outfeatures, -1]
+    """
+    def __init__(self):
+        super().__init__()
+
+        self.cnn = tvm.densenet121(pretrained=True)
+        self._load_from_ckpt()
+        self.out_features = self.cnn.classifier.in_features
+
+    def _load_from_ckpt(self, ckpt='./model.pth.tar'):
+        # Only support the specific chekpoint
```

Comment:
what do you mean by only suppor specific checkpoint? And if that's the case, why even have a parameter here?

Reply:
Because we are using the pretrained weights here https://github.com/berneylin/chexnet

For different pretrained weights, the loading codes may be different. I have removed the parameter.